City-of-Helsinki/berth-reservations-admin
# Implement better handling of application and lease deletion

## Description :sparkles:

Introduce utilities for checking whether leases and applications can be deleted.
These are used in the different invoicing pages to determine whether it should be
possible for the user to delete leases or applications.

Additionally fixes VEN-889.

## Issues :bug:

[VEN-889](https://helsinkisolutionoffice.atlassian.net/browse/VEN-889)

## Testing :alembic:

### Automated tests :gear:️

- Updated relevant tests
- `src/common/utils/__tests__/applicationUtils.test.ts `
- `src/common/utils/__tests__/leaseUtils.test.ts `

### Manual testing :construction_worker_man:

In single application views:

It should only be possible to delete the application if the application's state is not one of:
- "Tarjous lähetetty"
- "Tarjous maksettu"
- "Vanhentunut"

It should only be possible to delete leases with drafted status.

## Additional notes :spiral_notepad:

Was unable to change `display` style of HDS button without `important` 🤔.
